### ISSUE TITLE
Knoppix doesn't boot on UEFI

### ISSUE THREAD

**Issue by makeworld-the-better-one** [ventoy/Ventoy#1042]
### Official FAQ

- [X] I have checked the official FAQ.

### Ventoy Version

1.0.49

### What about latest release

Yes. I have tried the latest release, but the bug still exist.

### BIOS Mode

UEFI Mode

### Partition Style

MBR

### Disk Capacity

32 GB

### Image file download link (if applicable)

http://torrent.unix-ag.uni-kl.de/

### What happened?

Transferred from [this](https://github.com/ventoy/Ventoy/issues/7#issuecomment-871721810) comment and the one below.

`KNOPPIX_V8.6.1-2019-10-14-EN.iso`, which is listed on the Ventoy website as tested and working on UEFI, does not boot. A blinking underscore that I believe is from Knoppix appears, but nothing happens after that. The same thing occurs with the newer version, `KNOPPIX_V9.1DVD-2021-01-25-EN.iso`.

I have not tested with Legacy BIOS.

**Comment by makeworld-the-better-one** [comment]
cc @mid-kid and @quantatheist

**Comment by ventoy** [comment]
I download `KNOPPIX_V9.1DVD-2021-01-25-EN.iso` with md5sum `f4cde43e70ff0e58d06e6f805251cd9c`
So the iso file is ok.
I directly connect the ISO file to the VMWare (UEFI mode) and boot. The VM crashed immediately. 
So I think the ISO file itself has some problem, it can NOT boot in UEFI mode from CDROM.

**Comment by ventoy** [comment]
The same issue with `KNOPPIX_V8.6-2019-08-08-EN.iso`
So I may made a mistake,  I have updated the test list in the website. [https://www.ventoy.net/en/distro_iso/79.html](https://www.ventoy.net/en/distro_iso/79.html)

**Comment by makeworld-the-better-one** [comment]
> The VM crashed immediately.

That's weird, as that's not what happened when I booted off Knoppix from Ventoy.

> So I think the ISO file itself has some problem, it can NOT boot in UEFI mode from CDROM.

According to the Knoppix website, UEFI boot is supported, see [here](https://knopper.net/knoppix/knoppix810-en.html#uefi).

[This page]() says:

> In UEFI boot mode there is no "text mode" screen. Because of this, the usual Knoppix boot messages are not shown, not even an eventually activated password prompt. In some cases, the computer even starts with a comletely "black screen" with no visible activity for a while, please don't give up too early! Only after the point where the kernel module serving the graphics adapter has been loaded successfully, the screen will light up again and the graphical user interface should start. This is a known problem when booting in UEFI secure mode.

I am not booting in secure mode, but maybe this is the issue! Maybe I just wasn't waiting long enough for Knoppix to boot up.

**Comment by mid-kid** [comment]
DDing the iso straight to a USB drive works for me, even with UEFI. I've used the iso like that sometimes, but the stick I have ventoy on is too slow to make writing on it any useful at all.

**Comment by makeworld-the-better-one** [comment]
Alright, so UEFI does work with Knoppix as advertised, there was just some issue with @ventoy's VM. I will try waiting much longer for Knoppix (like 10 mins) and report back.

**Comment by ventoy** [comment]
@mid-kid   DD the ISO to the USB to boot is much different from boot from Ventoy.
Ventoy enulate the ISO file as a CDROM to boot.

So it is similar to burn the ISO file to an empty CD/DVD and use a physical CDROM to boot CD/DVD

**Comment by ventoy** [comment]
Anyway, if you connect the ISO file directly to a VM(e.g. VMware/VirtualBox) and it can not boot fine in UEFI mode.
That means the ISO file has some problem.
Many other iso files (e.g. Ubuntu/CentOS/Arch/......)  can boot fine in the same way.

**Comment by mid-kid** [comment]
Apparently, you're right. According to [the release notes](http://www.knopper.net/knoppix/knoppix910-en.html):
```
KNOPPIX 9.1 generally supports booting in → UEFI and → Secure Boot after installation on USB flash disk. But in order to use this feature, an entry in the UEFI firmware is necessary at first boot attempt. → Find quick instructions here.
```

Knoppix only supports UEFI when flashed onto a USB drive. I'm not sure *why* this would be a thing, since the data on both mediums would be the same, but apparently it is. Can't ventoy support such images?

**Comment by quantatheist** [comment]
I don't know much about this, but may the problem have something to do with Knoppix' hybrid layout? (https://www.knopper.net/knoppix/knoppix861-en.html#layout - section "partition layouts")

**Comment by ventoy** [comment]
Anyway, Knoppix should not make it so special and different from most of other distros(e.g. Ubuntu/CentOS/Debian/Arch...)

**Comment by mid-kid** [comment]
Booting it in a UEFI VM isn't an issue, you just have to pretend it's a USB or hard disk image:
```
qemu-system-x86_64 -m 2G -enable-kvm -hda KNOPPIX_V9.1DVD-2021-01-25-EN.iso -bios /usr/share/edk2-ovmf/OVMF_CODE.fd
```

> Knoppix should not make it so special and different from most of other distros

I'll try to ask, but even if I did I'm sure there's some technical/legacy reason behind this. Knoppix isn't really a "normal" livecd.

**Comment by makeworld-the-better-one** [comment]
Thanks @mid-kid! Good to see you could get UEFI working in some way.

I still think the issue might be what I mentioned [above](https://github.com/ventoy/Ventoy/issues/1042#issuecomment-892621890), that we just haven't waiting long enough for the bootup. I'll report back when I get a chance to try Knoppix again. If anyone else can try that'd be good too.

**Comment by ventoy** [comment]
Some distro (e.g. Tails) provide both  `.iso` and `.img`. 
So since knoppis provides `.iso` it should  boot fine as an iso file not as a hard disk img.

**Comment by mid-kid** [comment]
You mean that if we rename it to .img it should work?

**Comment by ventoy** [comment]
NO.  .img and .iso is in different format.
So in current case, the KNOPPIX can not boot with Ventoy in UEFI mode.

**Comment by makeworld-the-better-one** [comment]
The extension shouldn't matter. Maybe they should have used a different one, but in any case Ventoy still can't make it work.

What can be done to move this forward? Clearly Knoppix *is* a special case, but it would be very useful to have it work.

**Comment by makeworld-the-better-one** [comment]
@ventoy any update on this? Thank you.

**Comment by ventoy** [comment]
@makeworld-the-better-one 

1. Use this CI release:  https://github.com/ventoy/Ventoy/actions/runs/3663991017
https://www.ventoy.net/en/doc_github_ci.html

2. Select `Boot in grub2 mode` when you boot Knoppix in UEFI mode.

**Comment by popy2k14** [comment]
@ventoy Knoppix 9.1 boots in grub2 mode, thx

**Comment by makew0rld** [comment]
I'm not about to test it, but if Knoppix is working with the latest Ventoy release, then it sounds like this issue can be closed.

**Comment by popy2k14** [comment]
Yes it boots in grub2 mode. Would be nice if it could also boot in default mode.
